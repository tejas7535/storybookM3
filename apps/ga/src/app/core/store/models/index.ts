export * from './bearing-selection-state.models';
export * from './calculation-parameters-state.models';
export * from './calculation-result-state.models';
export * from './router-state.models';
export * from './settings-state.models';
