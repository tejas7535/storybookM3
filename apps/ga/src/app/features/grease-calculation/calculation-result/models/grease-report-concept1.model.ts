import {
  GreaseReportSubordinate,
  GreaseReportSubordinateTitle,
} from './grease-report-subordinate.model';

export const CONCEPT1 = 'CONCEPT1';

export enum SUITABILITY {
  'YES' = 'LB_YES',
  'NO' = 'LB_NO',
  'CONDITION' = 'LB_CONDITION',
  'UNKNOWN' = 'LB_UNKNOWN',
}

export enum SUITABILITY_LABEL {
  'SUITED' = 'SUITED',
  'NOT_SUITED' = 'NOT_SUITED',
  'UNSUITED' = 'UNSUITED',
  'CONDITIONAL' = 'CONDITIONAL',
  'UNKNOWN' = 'UNKNOWN',
}

export enum CONCEPT1_SIZES {
  '60ML' = 60,
  '125ML' = 125,
}

export type GreaseReportConcept1Subordinate = Omit<
  GreaseReportSubordinate,
  'titleID'
> & {
  titleID: `${GreaseReportSubordinateTitle}` | `${SUITABILITY}`;
};
