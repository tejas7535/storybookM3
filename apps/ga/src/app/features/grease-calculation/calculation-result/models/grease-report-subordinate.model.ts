import { GreaseResult } from './grease-result.model';

export type GreaseReportSubordinateIdentifier =
  | 'text'
  | 'textPairList'
  | 'textBlock'
  | 'block'
  | 'table'
  | 'legalNote'
  | 'variableBlock'
  | 'variableLine'
  | 'greaseResult';

export enum GreaseReportSubordinateTitle {
  STRING_OUTP_INPUT = 'STRING_OUTP_INPUT',
  STRING_OUTP_RESULTS = 'STRING_OUTP_RESULTS',
  STRING_OUTP_RESULTS_FOR_GREASE_SELECTION = 'STRING_OUTP_RESULTS_FOR_GREASE_SELECTION',
  STRING_OUTP_RESULTS_FOR_GREASE_SERVICE_STRING_OUTP_GREASE_QUANTITY_IN_CCM = 'STRING_OUTP_RESULTS_FOR_GREASE_SERVICE_STRING_OUTP_GREASE_QUANTITY_IN_CCM',
  STRING_OUTP_OVERVIEW_OF_CALCULATION_DATA_FOR_GREASES = 'STRING_OUTP_OVERVIEW_OF_CALCULATION_DATA_FOR_GREASES',
}

export enum GreaseReportSubordinateHint {
  ERRORS = 'errors',
  WARNINGS = 'warnings',
  NOTES = 'notes',
}

export enum SubordinateDataItemField {
  GREASE_GRADE = 'Grease grade',
  BASEOIL = 'BaseOil',
  NLGI = 'NLGI',
  THICKENER = 'Thickener',
  QVIN = 'Qvin',
  QVRE_MAN_MIN = 'Qvre_man_min',
  QVRE_MAN_MAX = 'Qvre_man_max',
  QVRE_AUT_MIN = 'Qvre_aut_min',
  QVRE_AUT_MAX = 'Qvre_aut_max',
  TFR_MIN = 'tfR_min',
  TFR_MAX = 'tfR_max',
  NY40 = 'ny40',
  T_LIM_LOW = 'T_lim_low',
  T_LIM_UP = 'T_lim_up',
  TFG_MIN = 'tfG_min',
  TFG_MAX = 'tfG_max',
  ADD_REQ = 'add_req',
  ADD_W = 'add_w',
  RHO = 'rho',
  F_LOW = 'f_low',
  VIP = 'vib',
  SEAL = 'seal',
  NSF_H1 = 'NSF-H1',
  KAPPA = 'kappa',
}

export interface GreaseReportSubordinateDataItem {
  value?: string | null | number;
  unit?: string;
  field?: `${SubordinateDataItemField}`;
}

export interface GreaseReportSubordinateDescription {
  identifier: string;
  title: string;
  entries: string[][];
}

export interface GreaseReportSubordinateData {
  originalFields?: string[];
  fields: string[];
  unitFields: (null | undefined | { unit: string })[];
  items: GreaseReportSubordinateDataItem[][];
}

export interface GreaseReportSubordinate {
  identifier: GreaseReportSubordinateIdentifier;
  title?: string;
  entries?: string[][];
  text?: string[];
  designation?: string;
  value?: string;
  data?: GreaseReportSubordinateData;
  abbreviation?: string;
  titleID?: `${GreaseReportSubordinateTitle}`;
  legal?: string;
  unit?: string;
  description?: GreaseReportSubordinateDescription;
  subordinates?: GreaseReportSubordinate[];
  defaultOpen?: boolean;
  content?: any;
  clickHandler?: any;
  greaseResult?: GreaseResult;
}
