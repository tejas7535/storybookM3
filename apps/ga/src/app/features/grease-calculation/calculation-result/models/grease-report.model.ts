import { GreaseReportSubordinate } from './grease-report-subordinate.model';

export interface CompanyInformation {
  url: string;
  company: string;
}

export interface GreaseReport {
  identifier: string;
  programName: string;
  programNameID: string;
  isBeta: boolean;
  method: string;
  methodID: string;
  title: string;
  titleID: string;
  subordinates: GreaseReportSubordinate[];
  companyInformation: CompanyInformation[];
  timeStamp: string;
  programVersion: string;
  transactionFileName: string;
}
