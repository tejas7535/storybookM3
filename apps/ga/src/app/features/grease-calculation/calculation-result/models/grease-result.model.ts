export interface GreaseResultDataItem {
  title: string;
  tooltip?: string;
  values?: string;
  custom?: {
    selector: string;
    data?: any;
  };
}

export type GreaseResultDataSourceItem = GreaseResultDataItem | undefined;

export type GreaseResultData = GreaseResultDataSourceItem[];

export interface PreferredGreaseResult {
  id: string;
  text: string;
}

export interface GreaseResult {
  mainTitle: string;
  subTitle?: string;
  isSufficient: boolean;
  isPreferred?: boolean;
  dataSource: GreaseResultData;
}
