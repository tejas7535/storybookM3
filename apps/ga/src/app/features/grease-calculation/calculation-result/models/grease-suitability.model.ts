import { SUITABILITY_LABEL } from './grease-report-concept1.model';

export enum GreaseSuitabilityLevels {
  'ExtremelySuitable' = '++',
  'HighlySuitable' = '+',
  'Suitable' = '0',
  'LessSuitable' = '-',
  'NotSuitable' = '--',
}

export type GreaseSuitabilityLevelsSetting = {
  [key in `${GreaseSuitabilityLevels}`]: keyof typeof GreaseSuitabilityLevels;
};

export interface GreaseConcep1Suitablity {
  hint: string;
  label: SUITABILITY_LABEL;
  c1_60: number;
  c1_125: number;
}
