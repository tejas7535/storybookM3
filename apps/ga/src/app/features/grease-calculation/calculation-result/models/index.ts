export * from './grease-report.model';
export * from './grease-report-concept1.model';
export * from './grease-report-subordinate.model';
export * from './grease-result.model';
export * from './grease-suitability.model';
export * from './pdf/grease-pdf-input.model';
export * from './pdf/grease-pdf-message.model';
export * from './pdf/grease-pdf-report.model';
export * from './pdf/grease-pdf-result.model';
export * from './result-section.model';
