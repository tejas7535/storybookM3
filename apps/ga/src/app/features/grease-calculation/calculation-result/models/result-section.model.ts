import { LabelValue } from '@schaeffler/label-value';

import {
  GreaseSelectionResult,
  InitialLubricationResult,
  PerformanceResult,
  RelubricationResult,
} from './grease-result.model';

export interface ResultSection {
  title: string;
  mainValue?: string;
  badgeText: string;
  badgeSecondaryText?: string;
  badgeClass?: string;
  concept1: boolean;
  extendable: boolean;
  labelValues?: LabelValue[];
}

export enum ResultSectionType {
  'INITIAL_LUBRICATION' = 'initialLubrication',
  'PERFORMANCE' = 'performance',
  'RELUBRICATION' = 'relubrication',
  'GREASE_SELECTION' = 'greaseSelection',
}

export type ResultSectionData = {
  [key in ResultSectionType]: ResultSection;
};

export type ResultSectionRaw =
  | InitialLubricationResult
  | PerformanceResult
  | RelubricationResult
  | GreaseSelectionResult;

export type InitialResultSectionData = {
  [key in ResultSectionType]: Partial<ResultSection>;
};
