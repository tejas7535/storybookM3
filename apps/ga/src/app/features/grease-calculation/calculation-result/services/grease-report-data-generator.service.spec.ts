import {
  createServiceFactory,
  mockProvider,
  SpectatorService,
} from '@ngneat/spectator/jest';
import { TranslocoModule } from '@ngneat/transloco';
import { TranslocoLocaleService } from '@ngneat/transloco-locale';

import { provideTranslocoTestingModule } from '@schaeffler/transloco/testing';

import {
  GREASE_PDF_INPUT_MOCK,
  GREASE_PDF_MESSAGE,
  GREASE_RESULT_SUBORDINATES_MOCK,
} from '@ga/testing/mocks';
import { GREASE_PDF_RESULT_MOCK } from '@ga/testing/mocks/models/pdf/grease-pdf-result.mock';

import { GreasePdfInput, GreasePdfMessage, GreasePdfResult } from '../models';
import { GreaseReportDataGeneratorService } from './grease-report-data-generator.service';

jest.mock('@ngneat/transloco', () => ({
  ...jest.requireActual<TranslocoModule>('@ngneat/transloco'),
  translate: jest.fn((translateKey: string) => {
    switch (translateKey) {
      default:
        return translateKey.toString().replace('calculationResult.', '');
    }
  }),
}));

describe('GreaseReportDataGeneratorService', () => {
  let spectator: SpectatorService<GreaseReportDataGeneratorService>;
  let service: GreaseReportDataGeneratorService;
  let automaticallyLubricated = true;
  const localizeNumber = jest.fn((number) => `${number}`);

  const createService = createServiceFactory({
    service: GreaseReportDataGeneratorService,
    imports: [provideTranslocoTestingModule({ en: {} })],
    providers: [mockProvider(TranslocoLocaleService, { localizeNumber })],
  });

  beforeEach(() => {
    spectator = createService();
    service = spectator.service;
  });

  it('should be created', () => {
    expect(service).toBeTruthy();
  });

  describe('get disclaimer title', () => {
    it('should get translated title', () => {
      const dislaimerTitle = service.getDisclaimerTitle();

      expect(dislaimerTitle).toBe('legal.disclaimer');
    });
  });

  describe('prepare report input data', () => {
    it('should return formatted data', () => {
      const result: GreasePdfInput = service.prepareReportInputData(
        GREASE_RESULT_SUBORDINATES_MOCK
      );

      expect(result).toEqual(GREASE_PDF_INPUT_MOCK);
    });

    it('should gracefully return empty data if empty input provided', () => {
      const result: GreasePdfInput = service.prepareReportInputData([]);

      expect(result).toEqual({
        sectionTitle: GREASE_PDF_INPUT_MOCK.sectionTitle,
        tableItems: [],
      } as GreasePdfInput);
    });
  });

  describe('prepare report result data', () => {
    it('should return formatted data', () => {
      const result: GreasePdfResult = service.prepareReportResultData(
        GREASE_RESULT_SUBORDINATES_MOCK,
        automaticallyLubricated
      );

      expect(result).toEqual(GREASE_PDF_RESULT_MOCK);
    });

    it('should gracefully return empty data if empty input provided', () => {
      const result: GreasePdfResult = service.prepareReportResultData(
        [],
        automaticallyLubricated
      );

      expect(result).toEqual({
        sectionTitle: '',
        tableItems: [],
      } as GreasePdfResult);
    });
  });

  describe('prepare report result data for not autolubrciated configuration', () => {
    beforeEach(() => {
      automaticallyLubricated = false;
    });

    it('should exclude concept1 information from result', () => {
      const result: GreasePdfResult = service.prepareReportResultData(
        GREASE_RESULT_SUBORDINATES_MOCK,
        automaticallyLubricated
      );

      expect(result).toEqual({
        sectionTitle: GREASE_PDF_RESULT_MOCK.sectionTitle,
        tableItems: [
          {
            title: GREASE_PDF_RESULT_MOCK.tableItems[0].title,
            subTitle: GREASE_PDF_RESULT_MOCK.tableItems[0].subTitle,
            items: GREASE_PDF_RESULT_MOCK.tableItems[0].items,
            concept1: undefined,
          },
          {
            title: GREASE_PDF_RESULT_MOCK.tableItems[1].title,
            subTitle: GREASE_PDF_RESULT_MOCK.tableItems[1].subTitle,
            items: GREASE_PDF_RESULT_MOCK.tableItems[1].items,
            concept1: undefined,
          },
          {
            title: GREASE_PDF_RESULT_MOCK.tableItems[2].title,
            subTitle: GREASE_PDF_RESULT_MOCK.tableItems[2].subTitle,
            items: GREASE_PDF_RESULT_MOCK.tableItems[2].items,
            concept1: undefined,
          },
        ],
      });
      expect(localizeNumber).toBeCalled();
    });
  });

  describe('prepare report errors and warnings data', () => {
    it('should return formatted data', () => {
      const result: GreasePdfMessage =
        service.prepareReportErrorsAndWarningsData(
          GREASE_RESULT_SUBORDINATES_MOCK
        );

      expect(result).toEqual(GREASE_PDF_MESSAGE);
    });

    it('should gracefully return empty data if empty input provided', () => {
      const result: GreasePdfMessage =
        service.prepareReportErrorsAndWarningsData([]);

      expect(result).toEqual({
        sectionTitle: 'calculationResult.errorsWarningsNotes',
        messageItems: [],
      } as GreasePdfMessage);
    });
  });
});
