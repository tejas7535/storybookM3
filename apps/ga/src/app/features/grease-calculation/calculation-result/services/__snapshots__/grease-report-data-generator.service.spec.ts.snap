// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GreaseReportDataGeneratorService prepare report result data for not autolubrciated configuration should exclude concept1 information from result 1`] = `
Object {
  "sectionTitle": "calculationResult.resultsDefault",
  "tableItems": Array [
    Object {
      "concept1": undefined,
      "isRecommended": undefined,
      "items": Array [
        Object {
          "itemTitle": "calculationResult.initialGreaseQuantity",
          "items": Array [
            "1234567.9 g/1234567.9 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer365days",
          "items": Array [
            "450617279.8 g/365 days",
            "450617279.8 mock_unit/365 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationQuantityPer1000OperatingHours",
          "items": Array [
            "51440328.7 g/hours",
            "51440328.7 mock_unit/hours",
          ],
        },
        Object {
          "itemTitle": "calculationResult.greaseServiceLife",
          "items": Array [
            "~ 51440 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer30days",
          "items": Array [
            "37037036.7 g/30 days",
            "37037036.7 mock_unit/30 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer7days",
          "items": Array [
            "8641975.2 g/7 days",
            "8641975.2 mock_unit/7 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.viscosityRatio",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.baseOilViscosityAt40",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowerTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.upperTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.additiveRequired",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.effectiveEpAdditivation",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.density",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowFriction",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.suitableForVibrations",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.supportForSeals",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.h1Registration",
          "items": Array [
            "1234567.89",
          ],
        },
      ],
      "subTitle": "Mineral oil",
      "title": "Arcanol MULTI2",
    },
    Object {
      "concept1": undefined,
      "isRecommended": undefined,
      "items": Array [
        Object {
          "itemTitle": "calculationResult.initialGreaseQuantity",
          "items": Array [
            "1234567.9 g/1234567.9 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer365days",
          "items": Array [
            "450617279.8 g/365 days",
            "450617279.8 mock_unit/365 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationQuantityPer1000OperatingHours",
          "items": Array [
            "51440328.7 g/hours",
            "51440328.7 mock_unit/hours",
          ],
        },
        Object {
          "itemTitle": "calculationResult.greaseServiceLife",
          "items": Array [
            "~ 51440 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer30days",
          "items": Array [
            "37037036.7 g/30 days",
            "37037036.7 mock_unit/30 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer7days",
          "items": Array [
            "8641975.2 g/7 days",
            "8641975.2 mock_unit/7 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.viscosityRatio",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.baseOilViscosityAt40",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowerTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.upperTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.additiveRequired",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.effectiveEpAdditivation",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.density",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowFriction",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.suitableForVibrations",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.supportForSeals",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.h1Registration",
          "items": Array [
            "1234567.89",
          ],
        },
      ],
      "subTitle": "Mineral oil",
      "title": "Arcanol MULTI2",
    },
    Object {
      "concept1": undefined,
      "isRecommended": undefined,
      "items": Array [
        Object {
          "itemTitle": "calculationResult.initialGreaseQuantity",
          "items": Array [
            "1234567.9 g/1234567.9 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer365days",
          "items": Array [
            "450617279.8 g/365 days",
            "450617279.8 mock_unit/365 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationQuantityPer1000OperatingHours",
          "items": Array [
            "51440328.7 g/hours",
            "51440328.7 mock_unit/hours",
          ],
        },
        Object {
          "itemTitle": "calculationResult.greaseServiceLife",
          "items": Array [
            "~ 51440 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer30days",
          "items": Array [
            "37037036.7 g/30 days",
            "37037036.7 mock_unit/30 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer7days",
          "items": Array [
            "8641975.2 g/7 days",
            "8641975.2 mock_unit/7 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.viscosityRatio",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.baseOilViscosityAt40",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowerTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.upperTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.additiveRequired",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.effectiveEpAdditivation",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.density",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowFriction",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.suitableForVibrations",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.supportForSeals",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.h1Registration",
          "items": Array [
            "1234567.89",
          ],
        },
      ],
      "subTitle": "Mineral oil",
      "title": "Arcanol MULTI2",
    },
  ],
}
`;

exports[`GreaseReportDataGeneratorService prepare report result data should return formatted data 1`] = `
Object {
  "sectionTitle": "calculationResult.resultsDefault",
  "tableItems": Array [
    Object {
      "concept1": Object {
        "concept125ml": Object {
          "conceptTitle": "calculationResult.concept1settings.concept1Size",
          "notes": "",
          "settingArrow": "calculationResult.concept1settings.setArrowSetting",
        },
        "concept60ml": Object {
          "conceptTitle": "calculationResult.concept1settings.concept1Size",
          "notes": "",
          "settingArrow": "calculationResult.concept1settings.setArrowSetting",
        },
        "title": "calculationResult.concept1",
      },
      "isRecommended": undefined,
      "items": Array [
        Object {
          "itemTitle": "calculationResult.initialGreaseQuantity",
          "items": Array [
            "1234567.9 g/1234567.9 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer365days",
          "items": Array [
            "450617279.8 g/365 days",
            "450617279.8 mock_unit/365 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationQuantityPer1000OperatingHours",
          "items": Array [
            "51440328.7 g/hours",
            "51440328.7 mock_unit/hours",
          ],
        },
        Object {
          "itemTitle": "calculationResult.greaseServiceLife",
          "items": Array [
            "~ 51440 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer30days",
          "items": Array [
            "37037036.7 g/30 days",
            "37037036.7 mock_unit/30 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer7days",
          "items": Array [
            "8641975.2 g/7 days",
            "8641975.2 mock_unit/7 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.viscosityRatio",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.baseOilViscosityAt40",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowerTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.upperTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.additiveRequired",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.effectiveEpAdditivation",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.density",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowFriction",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.suitableForVibrations",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.supportForSeals",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.h1Registration",
          "items": Array [
            "1234567.89",
          ],
        },
      ],
      "subTitle": "Mineral oil",
      "title": "Arcanol MULTI2",
    },
    Object {
      "concept1": Object {
        "concept125ml": Object {
          "conceptTitle": "calculationResult.concept1settings.concept1Size",
          "notes": "not supported for greasing",
          "settingArrow": "",
        },
        "concept60ml": Object {
          "conceptTitle": "calculationResult.concept1settings.concept1Size",
          "notes": "not supported for greasing",
          "settingArrow": "",
        },
        "title": "calculationResult.concept1",
      },
      "isRecommended": undefined,
      "items": Array [
        Object {
          "itemTitle": "calculationResult.initialGreaseQuantity",
          "items": Array [
            "1234567.9 g/1234567.9 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer365days",
          "items": Array [
            "450617279.8 g/365 days",
            "450617279.8 mock_unit/365 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationQuantityPer1000OperatingHours",
          "items": Array [
            "51440328.7 g/hours",
            "51440328.7 mock_unit/hours",
          ],
        },
        Object {
          "itemTitle": "calculationResult.greaseServiceLife",
          "items": Array [
            "~ 51440 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer30days",
          "items": Array [
            "37037036.7 g/30 days",
            "37037036.7 mock_unit/30 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer7days",
          "items": Array [
            "8641975.2 g/7 days",
            "8641975.2 mock_unit/7 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.viscosityRatio",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.baseOilViscosityAt40",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowerTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.upperTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.additiveRequired",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.effectiveEpAdditivation",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.density",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowFriction",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.suitableForVibrations",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.supportForSeals",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.h1Registration",
          "items": Array [
            "1234567.89",
          ],
        },
      ],
      "subTitle": "Mineral oil",
      "title": "Arcanol MULTI2",
    },
    Object {
      "concept1": undefined,
      "isRecommended": undefined,
      "items": Array [
        Object {
          "itemTitle": "calculationResult.initialGreaseQuantity",
          "items": Array [
            "1234567.9 g/1234567.9 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer365days",
          "items": Array [
            "450617279.8 g/365 days",
            "450617279.8 mock_unit/365 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationQuantityPer1000OperatingHours",
          "items": Array [
            "51440328.7 g/hours",
            "51440328.7 mock_unit/hours",
          ],
        },
        Object {
          "itemTitle": "calculationResult.greaseServiceLife",
          "items": Array [
            "~ 51440 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer30days",
          "items": Array [
            "37037036.7 g/30 days",
            "37037036.7 mock_unit/30 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.relubricationPer7days",
          "items": Array [
            "8641975.2 g/7 days",
            "8641975.2 mock_unit/7 days",
          ],
        },
        Object {
          "itemTitle": "calculationResult.viscosityRatio",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.baseOilViscosityAt40",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowerTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.upperTemperatureLimit",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.additiveRequired",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.effectiveEpAdditivation",
          "items": Array [
            "1234567.89",
          ],
        },
        Object {
          "itemTitle": "calculationResult.density",
          "items": Array [
            "1234567.89 mock_unit",
          ],
        },
        Object {
          "itemTitle": "calculationResult.lowFriction",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.suitableForVibrations",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.supportForSeals",
          "items": Array [
            "0 (suitable)",
          ],
        },
        Object {
          "itemTitle": "calculationResult.h1Registration",
          "items": Array [
            "1234567.89",
          ],
        },
      ],
      "subTitle": "Mineral oil",
      "title": "Arcanol MULTI2",
    },
  ],
}
`;
